source/common/http/conn_pool_grid.h

```diff
@@ -38,6 +38,9 @@ class ConnectivityGrid : public ConnectionPool::Instance,
     public:
       ConnectionAttemptCallbacks(WrapperCallbacks& parent, PoolIterator it);
 
+      // Returns true if a stream is immediately created, false if it is pending.
+      bool newStream();
```

Comment:
It would be more readable to have an enum instead of a bool here. But you can leave it like this if you like it better. I got confused trying to read the code before seeing this comment.

Reply:
Sounds good - will do that in a follow-up!